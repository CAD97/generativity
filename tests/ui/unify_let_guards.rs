use generativity::{guard, Guard};

fn unify<'id>(_: Guard<'id>, _: Guard<'id>) {
    unreachable!()
}

fn main() {
    guard!(let a, b);
    unify(a, b);
}
