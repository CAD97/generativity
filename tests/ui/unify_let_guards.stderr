error[E0597]: `id` does not live long enough
  --> tests/ui/unify_let_guards.rs:8:5
   |
8  |     guard!(let a, b);
   |     ^^^^^^^^^^^^^^^^^ borrowed value does not live long enough
9  |     unify(a, b);
10 | }
   | -
   | |
   | `id` dropped here while still borrowed
   | borrow might be used here, when `brand` is dropped and runs the `Drop` code for type `Brand`
   |
   = note: values in a scope are dropped in the opposite order they are defined
   = note: this error originates in the macro `guard` (in Nightly builds, run with -Z macro-backtrace for more info)
